src/test/time/clock_getres/test_clock_getres.rs

```diff
@@ -0,0 +1,135 @@
+use test_utils::{set, FuzzArg, FuzzError, TestEnvironment, VerifyOrder};
+
+fn main() -> anyhow::Result<()> {
+    // should we restrict the tests we run?
+    let filter_shadow_passing = std::env::args().any(|x| x == "--shadow-passing");
+    let filter_libc_passing = std::env::args().any(|x| x == "--libc-passing");
+    // should we summarize the results rather than exit on a failed test
+    let summarize = std::env::args().any(|x| x == "--summarize");
+
+    let mut tests = get_tests();
+
+    if filter_shadow_passing {
+        tests.retain(|x| x.passing(TestEnvironment::Shadow));
+    }
+    if filter_libc_passing {
+        tests.retain(|x| x.passing(TestEnvironment::Libc));
+    }
+
+    test_utils::run_tests(&tests, summarize)?;
+
+    Ok(())
+}
+
+fn get_tests() -> Vec<test_utils::ShadowTest<(), anyhow::Error>> {
+    let mut tests: Vec<test_utils::ShadowTest<(), anyhow::Error>> = vec![];
+
+    let clockids = vec![
+        FuzzArg::new(libc::CLOCK_REALTIME, Ok(())),
+        FuzzArg::new(libc::CLOCK_REALTIME_ALARM, Ok(())),
+        FuzzArg::new(libc::CLOCK_REALTIME_COARSE, Ok(())),
+        FuzzArg::new(libc::CLOCK_TAI, Ok(())),
+        FuzzArg::new(libc::CLOCK_MONOTONIC, Ok(())),
+        FuzzArg::new(libc::CLOCK_MONOTONIC_COARSE, Ok(())),
+        FuzzArg::new(libc::CLOCK_MONOTONIC_RAW, Ok(())),
+        FuzzArg::new(libc::CLOCK_BOOTTIME, Ok(())),
+        FuzzArg::new(libc::CLOCK_BOOTTIME_ALARM, Ok(())),
+        FuzzArg::new(libc::CLOCK_PROCESS_CPUTIME_ID, Ok(())),
+        FuzzArg::new(libc::CLOCK_THREAD_CPUTIME_ID, Ok(())),
+        FuzzArg::new(
+            128,
+            Err(FuzzError::new(
+                VerifyOrder::First,
+                Some(-1),
+                Some(libc::EINVAL),
+            )),
+        ),
+        FuzzArg::new(
+            -1,
+            Err(FuzzError::new(
+                VerifyOrder::First,
+                Some(-1),
+                Some(libc::EINVAL),
+            )),
+        ),
+    ];
+
+    let resolutions: Vec<FuzzArg<*mut libc::timespec>> = vec![
+        FuzzArg::new(
+            &mut libc::timespec {
+                tv_sec: 0,
+                tv_nsec: 0,
+            },
+            Ok(()),
+        ),
+        FuzzArg::new(std::ptr::null_mut(), Ok(())),
+    ];
```

Comment:
I think this is UB. It's constructing a pointer to a temporary stack object, but this object will immediately go out of scope.

Reply:
Good find, thanks!